src/hooks/useLensProtocolName.ts

```diff
@@ -0,0 +1,36 @@
+import { ApolloClient, gql, InMemoryCache } from "@apollo/client"
+import { useWeb3React } from "@web3-react/core"
+import useSWRImmutable from "swr/immutable"
+
+const apolloClient = new ApolloClient({
+  uri: "https://api.lens.dev/",
+  cache: new InMemoryCache(),
+})
+
+const fetchLensProtocolName = (_, query) =>
+  apolloClient
```

Comment:
I think we don't really need the Apollo client here, we could make a simple `POST` requests with our `fetcher` instead. Here's an example:

https://github.com/agoraxyz/guild.xyz/blob/main/src/requirements/Snapshot/hooks/useSpace.ts#L5-L18